### ISSUE TITLE
Smashing railings on firetower causes floors to be built above

### ISSUE THREAD

**Issue by natsirt721** [CleverRaven/Cataclysm-DDA#64991]
### Describe the bug

Railings on fire towers do not have a roof above them. When the railing on the fire lookout tower is destroyed, it is replaced with a floor tile that adds a roof. When the terrain is loaded for the first time after smashing the railing, a floor is created one z level up.

### Attach save file

N/A

### Steps to reproduce

1. Find fire lookout tower
2. Go to the top room, smash railings
3. Observe railings converted to floor on the current z-level (as expected)
4. Go up one Z level, observe thin air above smashed railings (as expected)
5. Save and load game
6. Observe floors appearing above the smashed railings

### Expected behavior

Floors don't magically appear when I break railings

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2728 (21H2)
- Game Version: e084982 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Bionic Professions [package_bionic_professions],
    No Fungal Growth [no_fungal_growth]
]

### Additional context

I suspect this is because the `fill_ter` for that level of the firetower is `t_floor` instead of `t_floor_noroof`. So when the terrain gets smashed it is replaced with this. Interestingly the change is not apparent on the next z-level up until a save-load cycle occurs, but if you try to e.g. place a solar panel on the new floor it won't generate power.

While it would solve this issue, IDK if changing the `fill_ter` to `t_floor_noroof` would be ideal, since there is other furniture in the tower that should be roofed over. Ideally there would be some way to specify a 'terrain beneath' for each thing - maybe there is already but I haven't seen any way to do that in the docs.

**Comment by PatrikLundell** [comment]
If I recall correctly, each terrain type that can be smashed has an entry for what it converts into, so it would be railing being defined as turning into t_floor when smashed. I don't think there's any more logic than that, so there's no way to let terrain convert into something depending on other circumstances (e.g. ground at the ground level and floor without a ceiling higher up in a building, or maybe empty air).
If railing doesn't generate a ceiling on construction then a smashed railing shouldn't either, logically.

**Comment by PatrikLundell** [comment]
I agree with @NetSysFire's suggestion, provided there is (or would be) a way to ensure the furniture is immovable so it can't be pushed around.

**Comment by natsirt721** [comment]
Are there other instances of smashable terrain that should get this treatment as well? It seems like fences and other things could be in a similar situation (#48452), and converting them to immovable furniture feels like kind of a kludge. What about something like allowing the OMT file to specify 'terrain under' for things like this? Then you can layer stuff like flowers and grass, or fences on dirt/grass/rocks/gravel etc. And it gets rid of that annoying grab message.

**Comment by NetSysFire** [comment]
>Are there other instances of smashable terrain that should get this treatment as well?

Possibly. One would need to go through the terrain definitions and check each and every entry manually. Fixing the railing issue alone might be pretty tedious on its own already.

**Comment by natsirt721** [comment]
If it isn't too egregious, I'm going to investigate adding a field to the mapgen json that allows the override of `ter_set` for terrain in them.
